KarhooUISDK/Workers/VehicleRules/VehicleRulesProvider.swift

```diff
@@ -56,48 +59,89 @@ final class KarhooVehicleRulesProvider: VehicleRulesProvider {
+            vehicleRules?.rules.first { $0.ruleType == .fallback }
+        }
+        
+        var defaultTypeRule: VehicleImageRule? {
+            vehicleRules?.rules.first { $0.type == quote.vehicle.type && $0.ruleType == .typeDefault }
+        }
+        
+        var anyMatchingTagRule: VehicleImageRule? {
+            var rule: VehicleImageRule?
+            let rulesWithOneTag = vehicleRules?.rules.filter { $0.tags.count == 1 }
+            quote.vehicle.tags
+                .map { $0.lowercased() }
+                .forEach { quoteTag in
+                    guard rule == nil else { return }
+                    rule = rulesWithOneTag?.first { ruleWithOneTag in
+                        ruleWithOneTag.tags
+                            .map { $0.lowercased() }
+                            .contains(quoteTag.lowercased())
                     }
-                return rule
-            }
-            
-            let specificRule = vehicleRules?.rules.first { rule in
-                guard rule.ruleType == .specific else { return false }
-                let ruleTags = Set(rule.tags.map { $0.lowercased() })
-                let quoteVehicleTags = Set(quote.vehicle.tags.map { $0.lowercased() })
-                return rule.type.lowercased() == quote.vehicle.type.lowercased() && ruleTags == quoteVehicleTags
-            }
-            
-            var rule: VehicleImageRule? {
-                if VehicleType(rawValue: quote.vehicle.type.uppercased()) == .moto {
-                    return defaultTypeRule
-                } else {
-                    return (specificRule ?? anyMatchingTagRule) ?? (defaultTypeRule ?? fallbackRule)
                 }
+            return rule
+        }
+        
+        let specificRule = vehicleRules?.rules.first { rule in
+            guard rule.ruleType == .specific else { return false }
+            let ruleTags = Set(rule.tags.map { $0.lowercased() })
+            let quoteVehicleTags = Set(quote.vehicle.tags.map { $0.lowercased() })
+            return rule.type.lowercased() == quote.vehicle.type.lowercased() && ruleTags == quoteVehicleTags
+        }
+        
+        var rule: VehicleImageRule? {
+            if VehicleType(rawValue: quote.vehicle.type.uppercased()) == .moto {
+                return defaultTypeRule
+            } else {
+                return (specificRule ?? anyMatchingTagRule) ?? (defaultTypeRule ?? fallbackRule)
             }
-            
-            completion(rule)
         }
+        return rule
+    }
+    
+    private func findVehicleBadgeRule(
+        for quote: Quote,
+        from vehicleRules: VehicleImageRules?
+    ) -> BadgeImageRule? {
+        guard let vehicleImageRules = vehicleRules else {
+            return nil
+        }
+
+        func findRule(for tag: String) -> BadgeImageRule? {
+            vehicleImageRules.badges.first(where: { $0.tags.contains(tag)})
+        }
+
+        let economyTag = "economy"
```

Comment:
What do you think about creating an enum for these 3 values? Something like:
```
enum VehicleBadgeType: String {
    case economy
    case electric
    case hybrid
}
```
Alternatively, create an array of these tags and iterate through it instead of writing a new if for each case?